### ISSUE TITLE
Extract SourceControl subdirectory to gem?

### ISSUE THREAD

**Issue by bglusman** [whitesmith/rubycritic#64]
Hey @guilhermesimoes or whoever else is working on rubycritic these days!

I've continued working on [Debt Ceiling](https://github.com/bglusman/debt_ceiling) off and on the last year or so, and just finished a v1 of a big new feature...  It adds/requires additional git/source control dependency, and while I considered using rugged, I decided to look at what you guys did for git code and liked your source control abstraction as a light weight interface with no additional dependencies...  

For now I just copy pasted your code into my repo and built some new features on it, but I wondered if you would be open to/interested in making it into a new shared gem we can both rely on together? Or, it just occurred to me to see if anyone else already has a ruby source control abstraction, we might both be able to rely on this [scm](https://github.com/postmodern/scm) gem I just found, but my addition to your library would need to be ported to theirs, and I think will only work for git...   Curious for thoughts/reactions, the PR with my changes to your code are [here](https://github.com/bglusman/debt_ceiling/pull/29/files) starting with `base.rb`, but missing implementation for travel_to_commit for mercurial, and some test coverage.  Just an initial feeler while I ponder next steps.

**Comment by bglusman** [comment]
Also considering this one, which also supports bazzar... https://github.com/blackducksoftware/ohloh_scm

**Comment by pedrocarmona** [comment]
:+1:

**Comment by troessner** [comment]
>> For now I just copy pasted your code into my repo and built some new features on it, but I wondered if you would be open to/interested in making it into a new shared gem we can both rely on together?

I think that's a pretty cool idea. scm though seems dead and far from ready. Last commit 8 months ago and no proper docs.

**Comment by bglusman** [comment]
Yeah, I came to same conclusion looking at scm and one or two other options IIRC, debt ceiling 0.4.0 was released with the copy pasted scm classes from RC, and some additions mostly to support git-notes, whichI kind of regret (I'd been thinking it was an easy way to memoize analysis on a repo in CI, but I forgot CI tends to clone the repo fresh every time, so Redis or tmp files seem like a better memo strategy, though git notes is still in there as an option, but could remove it it makes sharing this easier)
